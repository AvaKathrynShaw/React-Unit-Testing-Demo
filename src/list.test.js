import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import List from './List';

it('should render a todo item ', () => {
  const tree = toJson(shallow(<List items={[]} />));
  expect(tree).toMatchSnapshot();
});

it('should match its snapshot with items', () => {
  const items = ['Learn react', 'rest', 'go out'];
  const tree = renderer.create(
    <List items={items} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
