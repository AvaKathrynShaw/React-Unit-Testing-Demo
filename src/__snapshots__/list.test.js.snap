// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot with items 1`] = `
<ul>
  <li>
    Learn react
  </li>
  <li>
    rest
  </li>
  <li>
    go out
  </li>
</ul>
`;

exports[`should render a todo item  1`] = `<ul />`;
